_source/_posts/2022-05-04-angular-microfrontend-deploy.md

```diff
@@ -0,0 +1,286 @@
+cd okta-angular-microfrontend-example
+git checkout local
+npm ci
+npm run run:all
+```
+_You'll need to follow the instructions on creating the Okta application and updating the code with your Okta domain and client ID._
+
+## Dynamic loading of micro-frontend remotes
+
+So far, we've been serving the cupcake micro-frontend website using the npm script `run:all`, which serves all three applications at once. Let's see what happens if we only try serving the host application. Stop serving the site, and run `ng serve shell` to serve the default application, `shell`, and open the browser.
+
+When you navigate to `http://localhost:4200`, you see a blank screen and a couple of console errors. The console errors happen because `shell` can't find the micro-frontend remotes at ports 4201 and 4202. 👀
+
+These errors mean the host is loading the micro-frontends upon initialization of the host, not upon route change like you might think based on configuring a lazy-loaded route in the routing module. Doh!
+
+With the `@angular-architects/module-federation` library's `loadRemoteModule()` method, we can dynamically load the micro-frontend remotes upon route change with a few quick changes.
+
+Open `projects/shell/src/app/app-routing.module.ts`.
+
+The function we pass into the `loadChildren` property for the lazy-loaded routes changes to using `loadRemoteModule()`. Update your routes array to match the code below.
+
+```ts
+import { loadRemoteModule } from '@angular-architects/module-federation';
+
+const routes: Routes = [
+  { path: '', component: ProductsComponent },
+  {
+    path: 'basket',
+    loadChildren: () => loadRemoteModule({
+      type: 'module',
+      remoteEntry: `http://localhost:4201/remoteEntry.js`,
+      exposedModule: './Module'
+    }).then(m => m.BasketModule)
+  },
+  {
+    path: 'profile',
+    loadChildren: () => loadRemoteModule({
+      type: 'module',
+      remoteEntry: `http://localhost:4202/remoteEntry.js`,
+      exposedModule: './Module'
+    }).then(m => m.ProfileModule)
+  },
+  { path: 'login/callback', component: OktaCallbackComponent }
+];
+```
+
+Now that you're loading the remotes within the route definitions, you no longer need to define the remote URI in the `projects/shell/webpack.config.js`. Go ahead and remove or comment out the remotes in the `webpack.config.js` file. Another benefit of dynamic remote loading is that we no longer need to declare the modules to help TypeScript. Feel free to delete the `projects/shell/decl.d.ts` file. 🪄
+
+Double-check everything still works by running `npm run run:all`.
+
+Then stop serving the project and try running only the `shell` application by running the following command:
+
+```shell
+ng serve shell
+```
+
+You should now see the shell application and only see a console error when navigating to a route served by a micro-frontend remote. Success!
+
+## Deploy your micro-frontend project using Vercel
```

Comment:
There are five paragraphs of text after this header. Maybe try to break it up by adding an image?